File: exam-system-java/entity/src/main/java/com/yyjy/exam/entity/question/entity/QuestionAnswers.java
package com.yyjy.exam.entity.question.entity;

import org.babyfish.jimmer.sql.*;
import org.jetbrains.annotations.Nullable;

import java.time.LocalDateTime;

@Entity
public interface QuestionAnswers {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	long id();
	
	@OneToOne
	@OnDissociate(DissociateAction.DELETE)
	Questions question();
	
	@IdView
	long questionId();
	
	String answer();
	
	@Nullable
	LocalDateTime createTime();
	
	@Nullable
	LocalDateTime updateTime();
	
	@Column(name = "deleted_at")
	@LogicalDeleted("now")
	@Nullable
	LocalDateTime delFlag();
}

File: exam-system-server/src/main/kotlin/com/yyjy/models/entity/QuestionAnswers.kt
package com.yyjy.models.entity

import org.babyfish.jimmer.sql.*
import java.time.LocalDateTime

/**
 * 题目答案表（标准答案和关键词）
 */
@Entity
interface QuestionAnswers {

    @Id
    @GeneratedValue(
        strategy = GenerationType.IDENTITY
    )
    val id: Long

    @OnDissociate(DissociateAction.DELETE)
    @OneToOne
    val question: Questions

    @IdView
    val questionId: Long

    val answer: String

    val keywords: String?

    /**
     * 创建时间
     */
    val createTime: LocalDateTime?

    /**
     * 更新时间
     */
    val updateTime: LocalDateTime?

    /**
     * 0-未删除，1-已删除
     */
    @Column(name = "deleted_at")
    @LogicalDeleted("now")
    val delFlag: LocalDateTime?
}

